src/ansys/simai/core/data/models.py

```diff
@@ -21,9 +21,85 @@
 # SOFTWARE.
 
 from dataclasses import asdict, dataclass
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Literal
+
+from pydantic import BaseModel, PositiveFloat, ValidationError, model_validator
 
 from ansys.simai.core.data.base import ComputableDataModel, Directory
+from ansys.simai.core.errors import InvalidArguments
+
+
+class DomainAxisDefinition(BaseModel):
+    """Representation of an axis in Domain of Analysis.
+
+    Args:
+        position: indicates the position of the anchor.
+
+                    | *relative_to_min*: it corresponds to `VolXmin = xmin - value`
+
+                    | *relative_to_max*: it corresponds to `VolXmax = xmax + value`
+
+                    | *relative_to_center*: it corresponds to `(xmin+xmax)/2 - value`
+
+                    | *absolute*: it corresponds to `VolXmin = value`
+
+        value:      the distance of the anchor from the position. When position is set to
+                    "absolute", value could be either positive or negative. Only positive
+                    values are accepted in any other case.
+        length:     the length of the Domain of Analysis along the axis.
+
+    """
+
+    position: Literal["relative_to_min", "relative_to_max", "relative_to_center", "absolute"]
+    value: float
+    length: PositiveFloat
+
+    @model_validator(mode="after")
+    def chek_value(self):
+        """Evaluates that the value is set correctly according to the position argument."""
+        if self.value < 0 and self.position != "absolute":
+            raise InvalidArguments(
+                "value should be a positive number when the position is not absolute"
+            )
+        return self
+
+
+@dataclass
+class DomainOfAnalysis:
+    """Definition of the Domain of Analysis.
+
+    Args:
```

Comment:
arguments in lower case: 

 Args:
        length: Domain of Analysis definition along the X axis.
        width: Domain of Analysis definition along the Y axis.
        height: Domain of Analysis definition along the Z axis.